### PR TITLE
common/intrusive_lru: Improve readability

### PR DESCRIPTION
**Pull request by Matan-B**
No change in current behavior. Only readability changes are introduced. (Comments, naming and etc.)

Signed-off-by: Matan Breizman <mbreizma@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by athanatos — APPROVED** [review]
needs a test run

File: src/common/intrusive_lru.h [hunk 1]
```diff
@@ -12,6 +12,13 @@ namespace ceph::common {
 /**
  * intrusive_lru: lru implementation with embedded map and list hook
  *
+ * Elements will be stored in an intrusive set.
+ * Once an element is no longer referenced it will
```
Comment (rzarzynski) [inline src/common/intrusive_lru.h:16]: nit: too aggressive line breaking? Around 78th column would be fine.

**Review by rzarzynski — APPROVED** [review]
Thanks for improving this!

Just tiny nits, not blockers for sure.

**Review by liu-chunmei — APPROVED** [review]

**Comment by Matan-B** [comment]
Crimson and rgw_bucket_sync_cache are the only 2 users, labeling accordingly for testing.

**Comment by Matan-B** [comment]
@cbodley, Is this approved by rgw?

**Comment by cbodley** [comment]
> @cbodley, Is this approved by rgw?

hey @Matan-B. the code changes look great, but i don't think i was able to get this through teuthology. i'll try again today!

**Comment by cbodley** [comment]
results in https://pulpito.ceph.com/cbodley-2023-03-06_21:14:14-rgw-wip-cbodley-testing-distro-default-smithi/ look good enough to merge for rgw. does this need any separate testing for crimson?

**Comment by Matan-B** [comment]
> results in https://pulpito.ceph.com/cbodley-2023-03-06_21:14:14-rgw-wip-cbodley-testing-distro-default-smithi/ look good enough to merge for rgw. does this need any separate testing for crimson?

Thank you! I'll run Crimson testing.

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2023-03-07_11:59:39-crimson-rados-wip-matanb-crimson-only-testing-7.3-distro-crimson-smithi/

**Pull request merged**